awx_collection/plugins/modules/tower_project.py

```diff
@@ -133,107 +143,129 @@
     tower_config_file: "~/tower_cli.cfg"
 '''
 
-from ..module_utils.ansible_tower import TowerModule, tower_auth_config, tower_check_mode
+from ..module_utils.tower_api import TowerModule
 
-try:
-    import tower_cli
-    import tower_cli.exceptions as exc
 
-    from tower_cli.conf import settings
-except ImportError:
-    pass
+def wait_for_project_update(module, last_request):
+    # The current running job for the udpate is in last_request['summary_fields']['current_update']['id']
 
+    if 'current_update' in last_request['summary_fields']:
+        running = True
+        while running:
+            result = module.get_endpoint('/project_updates/{}/'.format(last_request['summary_fields']['current_update']['id']))['json']
+
+            if module.is_job_done(result['status']):
+                running = False
```

Comment:
This should have at least a little delay in the loop. Otherwise, it'll occupy a uWSGI worker almost constantly.